### PR TITLE
Add functionality to limit the display of and delete comments

### PR DESCRIPTION
**Pull request by linamontes10-zz**
Add functionality that allows the user to decide the number of comments to display on the page with user input from a GET request.

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/AddCommentsServlet.java [hunk 1]
```diff
@@ -75,4 +79,27 @@ private String convertToJsonUsingGson(ArrayList<String> comments) {
...
+  private int getCommentLimit(HttpServletRequest request) {
+    String commentLimitString = request.getParameter("comment-limit");
+
+    int commentLimit;
+
+    try {
+      commentLimit = Integer.parseInt(commentLimitString);
+    } catch (NumberFormatException e) {
+      // Set default value for comment limit if parsing fails
+      return commentLimit = 10;
```
Comment (SarahGB) [inline portfolio/src/main/java/com/google/sps/servlets/AddCommentsServlet.java]: Nit: Several of these values (0, 10, 100) could be top level constants. 
Typically these are formatted as something like
`const DEFAULT_COMMENT_LIMIT = 10;`

File: portfolio/src/main/java/com/google/sps/servlets/DeleteCommentsServlet.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
+// Copyright 2019 Google LLC
```
Comment (SarahGB) [inline portfolio/src/main/java/com/google/sps/servlets/DeleteCommentsServlet.java:1]: Was this intended to be part of this PR? If so, please update the name/description to include this.

File: portfolio/src/main/webapp/index.html [hunk 3]
```diff
@@ -24,12 +23,16 @@ <h1 id="title">Lina's Portfolio</h1>
...
       <div id="fun-fact-container"></div>
-      <h1>Messages</h1>
-      <p>Click here to see messages from the server:</p>
-      <button onclick="getMessages()">Fetch server messages!</button>
-      <div id="messages-container"></div>
 
-      <form action="/data" method="POST">
+      <h1>Comments:</h1>
+      <label for="comment-limit">Enter the limit on the number of comments you want to display:</label>
+      <input type="number" id="comment-limit" name="comment-limit" min="0" max ="100" value = "10" onchange="displayComments()">
```
Comment (SarahGB) [inline portfolio/src/main/webapp/index.html]: Nit: Break this into multiple lines for readability.

File: portfolio/src/main/java/com/google/sps/servlets/DeleteCommentsServlet.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+/** Servlet that deletes comments data using Datastore */
+@WebServlet("/delete-comments")
+public class DeleteCommentsServlet extends HttpServlet {
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+    // Query is sorted in descending order to show most recent comment entitites first
```
Comment (SarahGB) [inline portfolio/src/main/java/com/google/sps/servlets/DeleteCommentsServlet.java]: It doesn't look like there is any ordering parameter here, so please update the comment.

File: portfolio/src/main/webapp/script.js [hunk 5]
```diff
@@ -37,18 +37,33 @@ function addRandomFunFact() {
...
+async function displayComments() {
+  const commentLimit = document.getElementById("comment-limit").value;
+  const response = await fetch(`/add-comments?comment-limit=${commentLimit}`);
+  const comments = await response.json();
+  let commentsResult;
+
+  if (comments.length) {
+    commentsResult = comments.join('\n\n');
+  } else {
+    commentsResult = "Uh-oh, there are no comments do display! You can add one below.";
```
Comment (SarahGB) [inline portfolio/src/main/webapp/script.js]: Nit: Fix typo in this sentence.

File: portfolio/src/main/webapp/style.css [hunk 6]
```diff
@@ -24,10 +43,43 @@ img {
...
 }
 
+textarea {
+  border: 1px solid #ccc;
+  border-radius: 4px;
+  box-sizing: border-box;
+  display: inline-block;
+  margin: 8px 0;
+  width: 100%;
+  padding: 12px 20px;
```
Comment (SarahGB) [inline portfolio/src/main/webapp/style.css]: Swap order of width and padding.

**Review by SarahGB — CHANGES_REQUESTED** [review]

File: portfolio/src/main/java/com/google/sps/servlets/AddCommentsServlet.java (hunk 1 above)
Comment (KeatonTech) [inline portfolio/src/main/java/com/google/sps/servlets/AddCommentsServlet.java]: You don't need to update commentLimit, and this syntax is a little funky. Just return 10

**Review by KeatonTech — CHANGES_REQUESTED** [review]

**Review by SarahGB — APPROVED** [review]

File: portfolio/src/main/webapp/index.html [hunk 7]
```diff
@@ -24,12 +23,18 @@ <h1 id="title">Lina's Portfolio</h1>
...
-      <h1>Messages</h1>
-      <p>Click here to see messages from the server:</p>
-      <button onclick="getMessages()">Fetch server messages!</button>
-      <div id="messages-container"></div>
 
-      <form action="/data" method="POST">
+      <h1>Comments:</h1>
+      <label for="comment-limit">Enter the limit on the number of comments you want to display:</label>
+      <input type="number" id="comment-limit" name="comment-limit"
+        min="0" max ="100" value = "10"
```
Comment (KeatonTech) [inline portfolio/src/main/webapp/index.html]: nit: Be consistent with the spacing. Usually in HTML we don't put a space around the =

File: portfolio/src/main/webapp/script.js [hunk 8]
```diff
@@ -37,18 +37,33 @@ function addRandomFunFact() {
...
+  const comments = await response.json();
+  let commentsResult;
+
+  if (comments.length) {
+    commentsResult = comments.join('\n\n');
+  } else {
+    commentsResult = "Uh-oh, there are no comments to display! You can add one below.";
+  }
+
+  console.log(comments);
```
Comment (KeatonTech) [inline portfolio/src/main/webapp/script.js]: nit: Do you still need this?

**Review by KeatonTech — APPROVED** [review]

**Pull request merged**